### ISSUE TITLE
[Bug]: Using latest version of Swashbuckle, OpenAPI documentation is incorrect when using custom TryParse() binding with Minimal APIs

### ISSUE THREAD

**Issue by marcominerva** [domaindrivendev/Swashbuckle.AspNetCore#3105]
### Describe the bug

Using latest versions of Swashbuckle (from **6.7.0** to the last one) with .NET 8.0.10, I have the same issue described in https://github.com/dotnet/aspnetcore/issues/36412. If I revert to Swashbuckle.AspNetCore **6.6.2**, everything works as expected.

### Steps to reproduce

Use Swashbuckle.AspNetCore **6.7.0** or later and try the code at https://github.com/dotnet/aspnetcore/issues/36412.

### Swashbuckle.AspNetCore version

6.7.0 or later

### .NET Version

8.0.403

**Comment by martincostello** [comment]
I wonder if we need to do something similar to how they fixed it: https://github.com/dotnet/aspnetcore/pull/36526

**Comment by marcominerva** [comment]
@martincostello if you can give me an hint about what part of Swashbuckle to look at, I could try to give it a look. In this moment, this is a blocking issue for me, so I would be happy to contribute.

**Comment by martincostello** [comment]
To be honest, I don't know myself off the top of my head without going digging in the code. I would imagine whatever it is in is here somewhere: https://github.com/domaindrivendev/Swashbuckle.AspNetCore/tree/master/src/Swashbuckle.AspNetCore.SwaggerGen/SchemaGenerator

**Comment by marcominerva** [comment]
By the way, the same issue happens also using the new built-in generator provided by .NET 9.0 RC2. I have opened an issue on the ASP.NET Core repository: https://github.com/dotnet/aspnetcore/issues/58318.

**Comment by jgarciadelanoceda** [comment]
The issue may probably be caused by #2980, however it has to be rethinked because this was also preventing the base types like int,Enums,DateTimes to be reported as the actual type ( Not just a string)

**Comment by marcominerva** [comment]
@jgarciadelanoceda, can you explain the issue are you referring to about base types?

I have tried with:

```csharp
app.MapGet("/test", (int number, DateTime dateTime, Status status) => TypedResults.NoContent());

public enum Status
{
    Active,
    Disabled
}
```

And I obtain this result (that seems correct):

![image](https://github.com/user-attachments/assets/c45dbb0d-5f45-43fd-8735-9dabd33ebb2d)

**Comment by jgarciadelanoceda** [comment]
With the version 6.6.2 the enum was not being shown as Enum and just as string, I am reviewing it

**Comment by jgarciadelanoceda** [comment]
I have created a PR.. I showed that the #2532 fixed the issue #2733 and I applied with the latest bits.
Sorry for it and thanks for this issue
